packages/type/src/reflection/processor.ts

```diff
@@ -1400,6 +1400,11 @@ export class Processor {
                 return isExtendable(a, b) ? a : { kind: ReflectionKind.never };
             }
 
+            // two different primitives always return never
+            if (isPrimitive(a) && isPrimitive(b) && a.kind !== b.kind) {
```

Comment:
This'll be faster in 99% of cases
```suggestion
            if (a.kind !== b.kind && isPrimitive(a) && isPrimitive(b)) {
```